src/main.ts

```diff
@@ -145,6 +145,48 @@ export async function runAction(): Promise<any> {
         )
       )
     }
+
+    core.debug("Processing pipeline report...")
+    const failedMessage =
+      "Some pipeline actions have failed. Please check the pipeline report for details."
+    if (result && !result["passed"]) {
+      core.info(ansi.red(failedMessage))
+    }
+
+    if (
+      !Object.values(constants.EndStates).includes(executionGraph["status"])
+    ) {
+      core.setFailed(`Execution graph ${executionGraphId} has timed out.`)
```

Comment:
Here you have the same use case as above. The graph times out but you still need to upload artifacts and `setFailed` is going to stop the workflow.